app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/AddReviewFragment.kt

```diff
@@ -82,23 +79,31 @@ class AddReviewFragment : Fragment(R.layout.fragment_add_review) {
 
         ratingBar.setOnRatingBarChangeListener { _, float, _ ->
             val rating = ReviewRating.fromValue(float)
-            viewModel.setRating(rating)
+            addReviewViewModel.setRating(rating)
         }
 
         comment.addTextChangedListener(onTextChangedTextWatcher { text, _, _, _ ->
-            viewModel.setComment(text?.toString())
+            addReviewViewModel.setComment(text?.toString())
         })
 
         title.addTextChangedListener(onTextChangedTextWatcher { text, _, _, _ ->
-            viewModel.setTitle(text?.toString())
+            addReviewViewModel.setTitle(text?.toString())
         })
     }
 
     /**
      *  Adds the review to the database
      */
     private fun addReview() {
-        if (viewModel.submitReview(activityViewModel.id)) {
+        if (!auth.isLoggedIn()) {
```

Comment:
Wouldn't it be better for the user to be notified before, when it clicks the "+" button, instead of after it has written its entire review?

Reply:
Ok, I moved the check to the onResume() method of the fragment.